File: src/main/java/dev/choimory/member/api/member/v1/command/domain/request/CreateMemberRequest.java
package dev.choimory.member.api.member.v1.command.domain.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import lombok.Builder;
import org.springframework.web.multipart.MultipartFile;

/**
 * 회원가입 요청 객체입니다.
 */
@Builder(toBuilder = true)
public record CreateMemberRequest(
        @JsonProperty("email") @NotBlank String email, // 이메일
        @JsonProperty("password") @NotBlank String password, // 비밀번호
        @JsonProperty("nickname") @NotBlank String nickname, // 닉네임
        @JsonProperty("introduce") String introduce, // 소개글
        @JsonProperty("profile") MultipartFile profile // 프로필 파일
) {
}

File: src/main/kotlin/dev/choimory/member/api/member/v1/command/domain/request/CreateMemberRequest.kt
package dev.choimory.member.api.member.v1.command.domain.request

import jakarta.validation.constraints.NotBlank
import org.springframework.web.multipart.MultipartFile

data class CreateMemberRequest(
    @NotBlank
    val email: String,
    @NotBlank
    val password: String,
    @NotBlank
    val nickname: String,
    val introduce: String?,
    val profile: MultipartFile?,
)
